p2p/src/peers.rs

```diff
@@ -71,6 +73,32 @@ impl Peers {
 		apeer.clone()
 	}
 
+	// Update the dandelion relay
+	pub fn update_dandelion_relay(&self) {
+		let peers = self.outgoing_connected_peers();
+		match thread_rng().choose(&peers) {
+			Some(peer) => {
+				// Clear the map and add new relay
+				self.dandelion_relay.write().unwrap().clear();
+				self.dandelion_relay
```

Comment:
This takes 2 write locks in a row, without unlock. Does that actually work? Either way I'd still recommend taking the lock only once and assigning it to a variable.

Reply:
I *think* I fix it. Can you take a look?